trio_asyncio/_sync.py

```diff
@@ -139,6 +139,14 @@ def is_done(_):
             nonlocal result
 
             result = outcome.capture(future.result)
+            if isinstance(result, outcome.Error) and isinstance(
+                result.error, (SystemExit, KeyboardInterrupt)
```

Comment:
looking at the [exception hierarchy](https://docs.python.org/3/library/exceptions.html#exception-hierarchy), do you need to include `GeneratorExit` here too?

Reply:
No; this is trying to match logic in asyncio where specifically these two exceptions will come out of the event loop if raised by a task.

`asyncio/tasks.py`:
```
class Task(...):
    ...
    def __step_run_and_handle_result(self, exc):
        coro = self._coro
        try:
            if exc is None:
                # We use the `send` method directly, because coroutines
                # don't have `__iter__` and `__next__` methods.
                result = coro.send(None)
            else:
                result = coro.throw(exc)
        ...
        except (KeyboardInterrupt, SystemExit) as exc:
            super().set_exception(exc)
            raise
        except BaseException as exc:
            super().set_exception(exc)
        else:
            ...
```